modules/c/nitf/source/ImageIO.c

```diff
@@ -6493,8 +6493,8 @@ NITFPRIV(int) nitf_ImageIO_writeMasks(_nitf_ImageIO * nitf,
     else
         padCodeLength = 0;
 
-    ((nitf_Uint8 *) buffer)[9] = padCodeLength >> 8;
-    ((nitf_Uint8 *) buffer)[8] = padCodeLength & 0xff;
+    ((nitf_Uint8 *) buffer)[8] = padCodeLength >> 8;
+    ((nitf_Uint8 *) buffer)[9] = padCodeLength & 0xff;
```

Comment:
Oh I see, the big endian byte swap check was taken out the other day in #27 but it was actually indexed backwards as part of that "fix"?  So here's what I'd recommend.

1. Add two convenience functions that do 2 and 4 byte endian swapping and have `nitf_ImageIO_swapMaskHeader()` call those (these are the same kinds of things we have in `sys/Conf.h` but sadly we can't use them since that's C.
2. Add a check to say if it's a big endian system, swap pad code length with this function
3. Do the memcpy approach above

I think that'll make the code more legible.  If you go this route, double check that the value is correct on both big and little endian systems. :o)